File: practica2/c++/jumpif.cc
#include "jumpif.h"
#include <stack>
#include <iostream>
#include <string>

using namespace std;

jumpif::jumpif(int32_t param)
:	argumento(param)
{
}

void jumpif::ejecutar(stack<int32_t>& pila, int32_t& pc) const
{
	int32_t tmp = pila.top();
	pila.pop();
	if(tmp >= 0)
	{
		pc = argumento;
	}
	else pc++;
}

string jumpif::nombre() const
{
	return "jumpif " + to_string(argumento);
}

File: practica2/java/Jumpif.java
import java.util.Stack;

public class Jumpif extends Inst{
    protected int argumento;

    public Jumpif(int param){
        argumento = param;
    }
    @Override
        public int ejecutar(Stack<Integer> pila, int pc) {
            int tmp = pila.peek();
	       pila.pop();
	       if(tmp >= 0)
	       {
	           return argumento;
	       }
           else {
               pc++;
               return pc;
           }    
        }
        @Override
        public String nombre()
        {
            return "jumpif " + Integer.toString(argumento);
        }
}
